nodejs/aws-infra/experimental/ecs/taskDefinition.ts

```diff
@@ -195,38 +156,32 @@ export abstract class TaskDefinition extends pulumi.ComponentResource {
     }
 }
 
-function createRunFunction(
-        isFargate: boolean,
-        taskDefArn: pulumi.Output<string>,
-        containerToEnvironment: pulumi.Output<Record<string, aws.ecs.KeyValuePair[]>>) {
+export type RunTaskRequest = utils.Overwrite<awssdk.ECS.RunTaskRequest, {
+    /**
+     * The Cluster to run the Task in.
+     */
+    cluster: ecs.Cluster;
+    taskDefinition: never;
+    launchType: never;
+}>;
+
+function createRunFunction(isFargate: boolean, taskDefArn: pulumi.Output<string>) {
+    return function runTask(
+                params: RunTaskRequest,
+                callback?: (err: awssdk.AWSError, data: awssdk.ECS.Types.RunTaskResponse) => void,
+            ): awssdk.Request<awssdk.ECS.Types.RunTaskResponse, awssdk.AWSError> {
```

Comment:
aside: i like how you wrapped this to make the signature clear.  it's a constant problem with these super long sigs on how to do it in a readable manner.